lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1168,7 +1168,10 @@ protected function get_layout_styles( $block_metadata ) {
 
 		if ( isset( $block_metadata['name'] ) ) {
 			$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_metadata['name'] );
-			if ( ! block_has_support( $block_type, array( '__experimentalLayout' ), false ) ) {
+			if (
+				! block_has_support( $block_type, array( '__experimentalLayout' ), false ) &&
+				! block_has_support( $block_type, array( 'spacing', 'margin' ), false )
```

Comment:
Trying to wrap my head around what this condition is doing -  if layout is not supported, when does it matter whether margin is supported or not?

Reply:
Ah, I see the confusion here, I should add a comment to clarify.

This check is about what the current block supports rather than what the theme supports, so to support layout aware margins for blocks without layout support (e.g. heading, paragraph, image) we need to allow it to pass at this point. The check is primarily an early return so that for blocks that we know don't need any handling, we don't need to do any further lookups to theme settings, etc. Previously, that was just blocks that didn't support layout, but we now need the check to be layout + margin.

Further down in the function, we check `$has_block_gap_support` (that is, the theme has block gap support) before outputting the layout aware margin rules.

Reply:
Got it, thanks for clarifying!